### PR TITLE
Consistent iters logic

### PR DESCRIPTION
**Pull request by carmocca**
Same as for Lit-Parrot: https://github.com/Lightning-AI/lit-parrot/pull/79

### CODE REVIEW

**Review by lantiga — APPROVED** [review]
Good catch

File: finetune/adapter.py [hunk 1]
```diff
@@ -120,9 +121,9 @@ def train(
 
     for iter_num in range(max_iters):
 
-        if step_count <= warmup_steps:
+        if step_count <= warmup_iters:
```
[the remark this answers is not in the document]
Comment (carmocca) [reply finetune/adapter.py:124]: Good point. We could change `step_count` to `iter_count` here or `*_iters` to `*_steps` across the code.

I don't have a preference either way. Just wanted to be consistent instead of using steps and iters interchangeably.

[the remark this answers is not in the document]
Comment (awaelchli) [reply finetune/adapter.py:124]: Steps and iters are not used interchangeably. We cannot use "steps" everywhere and we cannot use "iters" everywhere, that's very confusing! Since we are using gradient accumulation, we need to keep track of the step count and learning rate scheduling only makes sense on the optimizer steps. So that is why we differentiate between steps and iters. The name `gradient_accumulation_steps` came from the nanoGPT repo.

[the remark this answers is not in the document]
Comment (carmocca) [reply finetune/adapter.py:124]: In that case, we can revert to the previous name. But then I would suggest that its formula and comment is removed from the adapter files, as it suggests that it is meant to track iterations (2 epochs) and not steps:

`2 * (epoch_size // micro_batch_size) // devices # 2 epochs`

`warmup_iters` also comes from [nanoGPT](https://github.com/karpathy/nanoGPT/blob/master/train.py#L229)
